File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/MainActivityBlock.java
package com.jaynewstrom.screenswitchersample;

import com.jaynewstrom.concrete.ConcreteBlock;

final class MainActivityBlock implements ConcreteBlock<MainActivityComponent> {
    private final ApplicationComponent applicationComponent;

    MainActivityBlock(ApplicationComponent applicationComponent) {
        this.applicationComponent = applicationComponent;
    }

    @Override public String name() {
        return getClass().getName();
    }

    @Override public MainActivityComponent createComponent() {
        return DaggerMainActivityComponent.builder()
                .applicationComponent(applicationComponent)
                .mainActivityModule(new MainActivityModule())
                .build();
    }
}

File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/MainActivityBlock.kt
package com.jaynewstrom.screenswitchersample

import com.jaynewstrom.concrete.ConcreteBlock

internal class MainActivityBlock(
    private val applicationComponent: ApplicationComponent
) : ConcreteBlock<MainActivityComponent> {
    override fun name(): String = javaClass.name

    override fun createComponent(): MainActivityComponent {
        return DaggerMainActivityComponent.builder()
            .applicationComponent(applicationComponent)
            .build()
    }
}
